src/mcp_agent/workflows/llm/multipart_converter_openai.py

```
@@ -0,0 +1,495 @@
+    @staticmethod
+    def _convert_image_content(content: ImageContent) -> ContentBlock:
+        """Convert ImageContent to OpenAI image_url content block."""
+        # Get image data using helper
+        image_data = get_image_data(content)
+
+        # OpenAI requires image URLs or data URIs for images
+        image_url = {"url": f"data:{content.mimeType};base64,{image_data}"}
+
+        # Check if the image has annotations for detail level
+        if hasattr(content, "annotations") and content.annotations:
+            if hasattr(content.annotations, "detail"):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)